Formula/ghc.rb

```diff
@@ -86,14 +105,27 @@ def install
 
       ENV.prepend_path "PATH", binary/"bin"
     end
+    resource("cabal-install").stage { (binary/"bin").install "cabal" }
+    system "cabal", "v2-update"
+
+    if build.head?
+      cabal_args = std_cabal_v2_args.reject { |s| s["installdir"] }
+      system "cabal", "v2-install", "alex", "happy", *cabal_args, "--installdir=#{binary}/bin"
+      system "./boot"
+    end
 
     system "./configure", "--prefix=#{prefix}", "--disable-numa", "--with-intree-gmp"
-    system "make"
-    ENV.deparallelize { system "make", "install" }
+    # Let hadrian handle its own parallelization
+    jobs = ENV.make_jobs
+    ENV.deparallelize do
+      system "./hadrian/build", "install", "-j#{jobs}",
+                                           "--prefix=#{prefix}",
+                                           "--flavour=release",
+                                           "--docs=no-sphinx-pdfs"
+    end
 
     bash_completion.install "utils/completion/ghc.bash" => "ghc"
-    (lib/"ghc-#{version}/package.conf.d/package.cache").unlink
-    (lib/"ghc-#{version}/package.conf.d/package.cache.lock").unlink
+    lib.glob("*/lib/package.conf.d/package.cache{,.lock}").map(&:unlink)
```

Comment:
Directory layout is odd maybe due to `hadrian`. There is now `lib/ghc-9.4.2/{bin,lib}` subdirectories. Don't know why.